packages/amazonq/test/e2e_new/amazonq/helpers/mcpHelper.ts

```diff
@@ -0,0 +1,308 @@
+        const filterActions = await body.findElement(By.css('.mynah-detailed-list-filter-actions-wrapper'))
+        const saveButton = await filterActions.findElement(
+            By.css('.mynah-button.fill-state-always.status-primary.mynah-ui-clickable-item')
+        )
+        await saveButton.click()
+        return true
+    } catch (e) {
+        console.error('Error saving the MCP server configuration:', e)
+        return false
+    }
+}
+
+export async function cancelMCPServerConfiguration(webviewView: WebviewView): Promise<boolean> {
+    try {
+        const sheetWrapper = await waitForElement(webviewView, By.id('mynah-sheet-wrapper'))
+        const body = await sheetWrapper.findElement(By.css('.mynah-sheet-body'))
+        const filterActions = await body.findElement(By.css('.mynah-detailed-list-filter-actions-wrapper'))
+        const saveButton = await filterActions.findElement(
+            By.css('.mynah-button.mynah-button-secondary.mynah-button-border.fill-state-always.mynah-ui-clickable-item')
+        )
+        await saveButton.click()
+        return true
+    } catch (e) {
+        console.error('Error saving the MCP server configuration:', e)
+        return false
+    }
+}
+
+/**
+ * Clicks the refresh button in the MCP server configuration panel
+ * @param webviewView The WebviewView instance
+ * @returns Promise<boolean> True if refresh button was found and clicked, false otherwise
+ */
+export async function clickMCPRefreshButton(webviewView: WebviewView): Promise<boolean> {
+    try {
+        const sheetWrapper = await waitForElement(webviewView, By.id('mynah-sheet-wrapper'))
+        const header = await sheetWrapper.findElement(By.css('.mynah-sheet-header'))
+        const actionsContainer = await header.findElement(By.css('.mynah-sheet-header-actions-container'))
+        const refreshButton = await actionsContainer.findElement(By.css('button:has(i.mynah-ui-icon-refresh)'))
+        await refreshButton.click()
+        return true
+    } catch (e) {
+        console.error('Error clicking the MCP refresh button:', e)
+        return false
+    }
+}
+
+/**
+ * Clicks the close/cancel button in the MCP server configuration panel
+ * @param webviewView The WebviewView instance
+ * @returns Promise<boolean> True if close button was found and clicked, false otherwise
+ */
+export async function clickMCPCloseButton(webviewView: WebviewView): Promise<boolean> {
+    try {
+        const sheetWrapper = await waitForElement(webviewView, By.id('mynah-sheet-wrapper'))
+        const header = await sheetWrapper.findElement(By.css('.mynah-sheet-header'))
+        const cancelButton = await header.findElement(By.css('button:has(i.mynah-ui-icon-cancel)'))
+        await webviewView.getDriver().executeScript('arguments[0].click()', cancelButton)
+        return true
+    } catch (e) {
+        console.error('Error closing the MCP overlay:', e)
+        return false
```

Comment:
general question on the pattern. Whats the advantage to converting result to a boolean, rather than throwing the error? 

My assumption is that if one of these steps fail, we want to immediately fail the test. (i.e. throw an error)